src/profile/helpers/editData.ts

```diff
@@ -103,9 +103,12 @@ export type EditFunctions = {
   updateData: (newProfileRoot: ProfileRoot) => boolean;
   updateAfterSavingError: (id: string) => boolean;
   resetItem: (targetItem: EditData) => boolean;
+  addItem: () => EditData;
+  hasNewItem: () => boolean;
+  removeItem: (targetItem: EditData) => Partial<FormValues> | null;
```

Comment:
Why do we wish to sometimes return partial FormValues and sometimes return null?

Reply:
When formValues are returned the caller knows it needs to save something. Null is returned when a new, unsaved, item is removed and there is no need for server call.